pyproject.toml

```diff
@@ -36,7 +37,7 @@ priority = "explicit"
 
 
 [build-system]
-requires = ["poetry-core>=1.0.0"]
+requires = ["poetry-core>=1.0.0", "cython<3"]
```

Comment:
any specific reason to restrict cython <3? This may create problems in future

Reply:
Added a comment. Currently some dependencies like econml requires this (see https://github.com/py-why/EconML/blob/e0271fdde8d94a70a9f67a07e1d0a5a8e964d3f5/pyproject.toml#L94)